frontend/src/pages/modelServing/screens/projects/ServingRuntimeModal/ServingRuntimeTemplateSection.tsx

```diff
@@ -108,35 +108,41 @@ const ServingRuntimeTemplateSection: React.FC<ServingRuntimeTemplateSectionProps
     ),
   }));
 
-  const getServingRuntime = () => (
-    <>
-      <MenuGroup
-        data-testid="project-scoped-serving-runtimes"
-        label={
-          <Flex
-            spaceItems={{ default: 'spaceItemsXs' }}
-            alignItems={{ default: 'alignItemsCenter' }}
-            style={{ paddingBottom: '5px' }}
+  const getServingRuntime = () => {
+    const filteredProjectScopedTemplates = filterProjectScopedTemplates!.filter((template) =>
+      getServingRuntimeDisplayNameFromTemplate(template)
+        .toLocaleLowerCase()
+        .includes(searchServingRuntime.toLocaleLowerCase()),
+    );
```

Comment:
Instead of null operator `!` it's better we use optional chaning `?` with fallback to empty array. As the null assertion might throw run time error when `filterProjectScopedTemplates` is undefined.

```suggestion
    const filteredProjectScopedTemplates = filterProjectScopedTemplates?.filter((template) =>
      getServingRuntimeDisplayNameFromTemplate(template)
        .toLocaleLowerCase()
        .includes(searchServingRuntime.toLocaleLowerCase()),
    ) || [];
```